### DISCUSSION TITLE
Controling transport variable while converging

### DISCUSSION THREAD

**Discussion by RichRoos** [su2code/SU2#1820] [Q&A]
Hey,

I have a question on the ability to influence the solution when the transport equations are being solved. I am trying to control the amplification factor for the eN transition model for the first few iterations. The gradients are still quite high at the start which result in high amplification factors and thus instant transition. You can also see this in the output as the calculated amplifications just jumps around above and below the critical value that should indicate transition.

I have been trying to check for large differences in amplification values inbetween iterations and see if I can reduce the solution when this happens as the thought is that, when solution transiens damp out, gradients will be smaller and thus this artificial damping effect will no longer be needed. The [paper](https://arc.aiaa.org/doi/10.2514/1.J052905) also talks about needed something like this. 

Now for the real question :): 

- Is there already a function meant for a situation like this? Something like an extreme under-relaxation or is it maybe possible to set an initial value for the amplification factor that is different then the Inf value set in the CTransENSolver constructor? Although I have some experience now with the code, it is quite vast... So probably I have not yet come across something that could help me with this problem. 

- Another question is on the best location to reculate the growth of the amplification factor. Could I place it as a Postprocessing step to loop over all the points again and reduce large values deviating from the mean produced from neighbouring points or perhaps directly after the source residual to save some calculation time (maybe using Solution_Old)?

**Comment by pcarruscag** [comment]
You can have a look at CTurbSASolver::ComputeUnderRelaxationFactor (you would implement the same for your new solver)
It limits solution updates to a fraction of the current values, that fraction is quite large for the SA model, it would probably need tuning in your case.

**Reply by RichRoos** [reply]
Thanks! That might work. I will give that a try.
